platforms/core-runtime/functional/src/main/java/org/gradle/internal/serialization/Cached.java

```diff
@@ -39,8 +46,8 @@ public static <T> Cached<T> of(Callable<T> computation) {
 
     private static class Deferred<T> extends Cached<T> implements java.io.Serializable, EvaluationOwner {
 
-        private Callable<T> computation;
-        private Try<T> result;
+        private volatile Callable<T> computation;
```

Comment:
❌ This class is not meant to be used in concurrent scenarios. 

If this is a temporary measure until the root cause is fixed, it should have a `TODO` attached to it.